### ISSUE TITLE
Unable to install Ki-nTree on Windows 10

### ISSUE THREAD

**Issue by crossband** [sparkmicro/Ki-nTree#132]
Hello There,
I am unable to install Ki-nTree on Windows 10. Aparently I have not installed Visual C++ correctly.

`python -m pip install kintree
Collecting kintree
  Using cached kintree-0.6.6-py3-none-any.whl (156 kB)
Requirement already satisfied: wrapt<2.0.0,>=1.14.1 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from kintree) (1.15.0)
Collecting multiprocess<0.71.0,>=0.70.14
  Using cached multiprocess-0.70.14-py38-none-any.whl (132 kB)
Collecting inventree<0.10.0,>=0.9.2
  Using cached inventree-0.9.2-py2.py3-none-any.whl (21 kB)
Collecting mouser<0.2.0,>=0.1.2
  Using cached mouser-0.1.2-py3-none-any.whl (7.0 kB)
Requirement already satisfied: PyYAML<6.0.0,>=5.4.1 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from kintree) (5.4.1)
Collecting validators<0.20.0,>=0.19.0
  Using cached validators-0.19.0-py3-none-any.whl
Requirement already satisfied: PySimpleGUI<5.0.0,>=4.60.4 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from kintree) (4.60.4)
Collecting digikey-api<2.0.0,>=1.0.0
  Using cached digikey_api-1.0.0-py3-none-any.whl (180 kB)
Collecting python-Levenshtein<0.13.0,>=0.12.2
  Using cached python-Levenshtein-0.12.2.tar.gz (50 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: fuzzywuzzy<0.19.0,>=0.18.0 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from kintree) (0.18.0)
Requirement already satisfied: retrying>=1.3.3 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from digikey-api<2.0.0,>=1.0.0->kintree) (1.3.4)
Collecting certauth>=1.3.0
  Using cached certauth-1.3.0-py2.py3-none-any.whl (10 kB)
Collecting inflection>=0.3.1
  Using cached inflection-0.5.1-py2.py3-none-any.whl (9.5 kB)
Requirement already satisfied: requests>=2.22.0 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from digikey-api<2.0.0,>=1.0.0->kintree) (2.28.0)
Requirement already satisfied: urllib3>=1.25.3 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from digikey-api<2.0.0,>=1.0.0->kintree) (1.26.9)
Collecting click<9.0,>=8.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting dill>=0.3.6
  Using cached dill-0.3.6-py3-none-any.whl (110 kB)
Requirement already satisfied: setuptools in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from python-Levenshtein<0.13.0,>=0.12.2->kintree) (49.2.1)
Collecting decorator>=3.4.0
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting pyopenssl
  Using cached pyOpenSSL-23.0.0-py3-none-any.whl (57 kB)
Collecting tldextract
  Using cached tldextract-3.4.0-py3-none-any.whl (93 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from requests>=2.22.0->digikey-api<2.0.0,>=1.0.0->kintree) (2022.6.15)
Requirement already satisfied: idna<4,>=2.5 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from requests>=2.22.0->digikey-api<2.0.0,>=1.0.0->kintree) (3.3)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from requests>=2.22.0->digikey-api<2.0.0,>=1.0.0->kintree) (2.0.12)
Requirement already satisfied: six>=1.7.0 in c:\users\maharramov\appdata\local\programs\python\python38\lib\site-packages (from retrying>=1.3.3->digikey-api<2.0.0,>=1.0.0->kintree) (1.16.0)
Collecting cryptography<40,>=38.0.0
  Using cached cryptography-39.0.1-cp36-abi3-win_amd64.whl (2.5 MB)
Collecting requests-file>=1.4
  Using cached requests_file-1.5.1-py2.py3-none-any.whl (3.7 kB)
Collecting filelock>=3.0.8
  Using cached filelock-3.9.0-py3-none-any.whl (9.7 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp38-cp38-win_amd64.whl (178 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Building wheels for collected packages: python-Levenshtein
  Building wheel for python-Levenshtein (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.8
      creating build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\StringMatcher.py -> build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\__init__.py -> build\lib.win-amd64-3.8\Levenshtein
      running egg_info
      writing python_Levenshtein.egg-info\PKG-INFO
      writing dependency_links to python_Levenshtein.egg-info\dependency_links.txt
      writing entry points to python_Levenshtein.egg-info\entry_points.txt
      writing namespace_packages to python_Levenshtein.egg-info\namespace_packages.txt
      writing requirements to python_Levenshtein.egg-info\requires.txt
      writing top-level names to python_Levenshtein.egg-info\top_level.txt
      reading manifest file 'python_Levenshtein.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*pyc' found anywhere in distribution
      warning: no previously-included files matching '*so' found anywhere in distribution
      warning: no previously-included files matching '.project' found anywhere in distribution
      warning: no previously-included files matching '.pydevproject' found anywhere in distribution
      writing manifest file 'python_Levenshtein.egg-info\SOURCES.txt'
      copying Levenshtein\_levenshtein.c -> build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\_levenshtein.h -> build\lib.win-amd64-3.8\Levenshtein
      running build_ext
      building 'Levenshtein._levenshtein' extension
      error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for python-Levenshtein
  Running setup.py clean for python-Levenshtein
Failed to build python-Levenshtein
Installing collected packages: python-Levenshtein, pycparser, inflection, filelock, dill, decorator, colorama, validators, requests-file, multiprocess, inventree, click, cffi, tldextract, mouser, cryptography, pyopenssl, certauth, digikey-api, kintree
  Running setup.py install for python-Levenshtein ... error
  error: subprocess-exited-with-error

  × Running setup.py install for python-Levenshtein did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running install
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.8
      creating build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\StringMatcher.py -> build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\__init__.py -> build\lib.win-amd64-3.8\Levenshtein
      running egg_info
      writing python_Levenshtein.egg-info\PKG-INFO
      writing dependency_links to python_Levenshtein.egg-info\dependency_links.txt
      writing entry points to python_Levenshtein.egg-info\entry_points.txt
      writing namespace_packages to python_Levenshtein.egg-info\namespace_packages.txt
      writing requirements to python_Levenshtein.egg-info\requires.txt
      writing top-level names to python_Levenshtein.egg-info\top_level.txt
      reading manifest file 'python_Levenshtein.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*pyc' found anywhere in distribution
      warning: no previously-included files matching '*so' found anywhere in distribution
      warning: no previously-included files matching '.project' found anywhere in distribution
      warning: no previously-included files matching '.pydevproject' found anywhere in distribution
      writing manifest file 'python_Levenshtein.egg-info\SOURCES.txt'
      copying Levenshtein\_levenshtein.c -> build\lib.win-amd64-3.8\Levenshtein
      copying Levenshtein\_levenshtein.h -> build\lib.win-amd64-3.8\Levenshtein
      running build_ext
      building 'Levenshtein._levenshtein' extension
      error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> python-Levenshtein

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.`

I installed Visual Studio Build Tools 2022. But I am not sure what else I require to install.

**Comment by eeintech** [comment]
Hello @crossband 

I was able to reproduce it, and here's the process I followed:
1. Navigate to https://visualstudio.microsoft.com/visual-cpp-build-tools/
2. Click on "Download Build Tools" button
3. In the installer, in the "Visual Studio Build Tools 2022" card, click "Modify"
![image](https://user-images.githubusercontent.com/4020546/222477482-6add633a-ba57-4562-a408-cdce4691e3a8.png)
4. Click on "Individual Components" tab then install the following components:
![image](https://user-images.githubusercontent.com/4020546/222477711-bf3dc163-7cfa-4862-98f1-958f20352bf9.png)
5. Re-install Ki-nTree
```shell
pip install - U kintree
```

It worked for me. However I know this is not ideal installation flow, I hate to see how complicated this single library makes things for you and others so I'm going to attempt to remove it for `1.0.0` version.
Thanks for bringing this to my attention, let me know if you have any other question.
